src/wast-parser.cc

```diff
@@ -2057,7 +2071,13 @@ Result WastParser::ParseBoundValueTypeList(TokenType token,
       bindings->emplace(name,
                         Binding(loc, binding_index_offset + types->size()));
       if (type.is_index()) {
-        types->push_back(Type(type.index()));
+        // TODO: Incorrect values can be misinterpreted by the parser.
+        if (type.index() >= static_cast<Index>(Type::Void)) {
```

Comment:
What does it mean to compare to `Type::Void` here?

Reply:
This is a short term workaround until #2562 lands. The index is used as reference index (0..`void`) or type (`void`..0xffffffff). The current code does not allow passing a reference at all, so making a test is impossible. Also converting any index to type is dangerous.

This code is used twice at the moment, and will be removed.

Reply:
Could we add a helper here to convey what we are trying to do ? Something like `type.isXXX()`?

Reply:
I can. However, this code should be dropped as soon as possible, since it is incorrect. Now a `(ref 0xffffffff)` is converted to `i32`.